Kernel-GL-HRK/gl-kernel-basecamp-2019
# Lesson05-07: Implement character device driver with procfs/sysfs interfaces

# Summary
This is my implementation of character device driver for text messaging between users. Such steps were followed during development:
- Created new Linux kernel module;
- Implemented read/write functions from file_operations struct;
- Provided buffer increase as a module parameter;
- Provided work only with ASCII strings;
- Implemented procfs interface to get information about current buffer size and capacity;
- Implemented sysfs interface for buffer clearance.
## Device driver testing scenario
1. Check if module is loaded and relevant device and interfaces exist.
2. Print information about initial buffer state from procfs inerface.
3. Write data to character device.
4. Check if info about written data available from procfs interface.
5. Read all data from character device.
6. Check if buffer is clear now.
7. Write data to device again and clear buffer with sysfs interface.
8. Try to write data blocks to character device from /dev/urandom to check if driver deny writing of non-ASCII strings.
